servers/src/grin/sync/header_sync.rs

```diff
@@ -307,4 +248,22 @@ mod test {
 			vec![10000, 9998, 9994, 9986, 9970, 9938, 9874, 9746, 9490, 8978, 7954, 5906, 1810, 0,]
 		);
 	}
+
+	#[test]
+	fn test_close_enough() {
```

Comment:
Not sure this test is enough. I add some log in `get_locator` function and seems the behavior is strange:
```
header_sync - sync: history locator: [1022, 511, 511, 511, 511, 511, 511, 511, 511, 511, 0]
```
Let me look into the detail of this `close_enough()` function.

Reply:
If I add this code into the test, I can reproduce this duplicated locators problem:
```

		// test with 1 history
		let heights: Vec<u64> = vec![64, 62, 58, 50, 34, 2, 0];
		let history_locator: Vec<(u64, Hash)> = vec![
			(0, zh.clone()),
		];
		let mut locator: Vec<(u64, Hash)> = vec![(0, zh.clone())];
		for h in heights {
			if let Some(l) = close_enough(&history_locator, h) {
				locator.push(l);
			}
		}
		println!("locator={:?}", locator);
```

Test output:
```
running 1 test
locator=[(0, 00000000)]
locator=[(0, 00000000), (0, 00000000)]
```

Reply:
You pulled my branch a little too early, I've added a few fixes for these issues soon after your first comment.